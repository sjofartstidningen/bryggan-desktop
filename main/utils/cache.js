'use strict';

const windows = new Map();

module.exports = { windows };
