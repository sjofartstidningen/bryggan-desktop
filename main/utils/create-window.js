'use strict';

const { BrowserWindow } = require('electron');
const ipc = require('electron-better-ipc');
const { windows } = require('./cache');
const { loadRoute } = require('./routes');

const defaultWindowOptions = {
  show: false,
};

function createWindow(name, windowOptions) {
  const isCreated = () => windows.has(name);
  const getWindow = () => windows.get(name);

  const create = () => {
    const win = new BrowserWindow(
      Object.assign({}, defaultWindowOptions, windowOptions),
    );

    windows.set(name, win);
    loadRoute(win, name);

    ipc.answerRenderer(`${name}-ready`, () => win.show());
  };

  const close = () => {
    if (isCreated()) {
      const win = getWindow();
      win.destroy();
      windows.delete(name);
    }
  };

  const hide = () => {
    if (isCreated()) {
      const win = getWindow();
      win.hide();
    }
  };

  const show = () => {
    if (isCreated()) {
      const win = getWindow();
      win.show();
    }
  };

  const toggle = () => {
    if (isCreated()) {
      close();
    } else {
      create();
    }
  };

  const toggleHidden = () => {
    if (!isCreated()) {
      create();
    } else {
      const win = getWindow();

      if (win.isVisible()) {
        hide();
      } else {
        show();
      }
    }
  };

  return {
    isCreated,
    getWindow,
    create,
    close,
    hide,
    show,
    toggle,
    toggleHidden,
  };
}

module.exports = { createWindow };
