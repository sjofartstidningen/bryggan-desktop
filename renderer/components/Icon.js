import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

const IconWrapper = styled.span`
  position: relative;
  display: inline-flex;
  align-self: center;
  width: 1em;
  height: 1em;
`;

const IconSvg = styled.svg`
  width: 1em;
  height: 1em;
  fill: none;
  stroke: currentColor;
  stroke-width: 2;
  stroke-linecap: round;
  stroke-linejoin: round;

  ${p =>
    p.baseline &&
    css`
      position: absolute;
      bottom: -0.125em;
    `};

  ${p =>
    p.useFill &&
    css`
      fill: currentColor;
      stroke: none;
    `};
`;

const IconProps = {
  baseline: PropTypes.bool,
  className: PropTypes.string,
};

const BaseIconProps = {
  ...IconProps,
  useFill: PropTypes.bool,
  children: PropTypes.node.isRequired,
};

const defaultProps = { baseline: true, className: '' };

function Icon({ baseline = true, useFill, className, children }) {
  return (
    <IconWrapper className={className}>
      <IconSvg
        xmlns="http://www.w3.org/2000/svg"
        width="24"
        height="24"
        viewBox="0 0 24 24"
        baseline={baseline}
        useFill={useFill}
      >
        {children}
      </IconSvg>
    </IconWrapper>
  );
}

Icon.propTypes = BaseIconProps;

function createIcon(elem) {
  const Comp = props => <Icon {...props}>{elem()}</Icon>;

  Comp.propTypes = IconProps;
  Comp.defaultProps = defaultProps;

  return Comp;
}

const Home = createIcon(() => (
  <Fragment>
    <path d="M3 9l9-7 9 7v11a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2z" />
    <polyline points="9 22 9 12 15 12 15 22" />
  </Fragment>
));

const Folder = createIcon(() => (
  <path d="M22 19a2 2 0 0 1-2 2H4a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h5l2 3h9a2 2 0 0 1 2 2z" />
));

const File = createIcon(() => (
  <Fragment>
    <path d="M13 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V9z" />
    <polyline points="13 2 13 9 20 9" />
  </Fragment>
));

const IdFile = createIcon(() => (
  <Fragment>
    <path d="M13 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V9z" />
    <path
      fill="currentColor"
      strokeWidth="0"
      d="M9.57,16c0,.08,0,.1-.1.1H8.32c-.08,0-.1,0-.1-.1V7.91c0-.07,0-.1.1-.1H9.48a.08.08,0,0,1,.09.09Z"
    />
    <path
      fill="currentColor"
      strokeWidth="0"
      d="M15.75,14.9v-7s0-.06-.08-.06H14.45a.07.07,0,0,0-.07.08v2H14a3,3,0,0,0-3.08,3.2h0c0,2.09,1.19,3.12,2.85,3.12a4.59,4.59,0,0,0,2-.45s.06,0,.06-.08A7.15,7.15,0,0,1,15.75,14.9Zm-1.37-3.8v3.8a1.53,1.53,0,0,1-.54.09c-.86,0-1.55-.57-1.55-2,0-1.24.67-2,1.64-2a1.13,1.13,0,0,1,.45.08Z"
    />
  </Fragment>
));

const Circle = createIcon(() => <circle cx="12" cy="12" r="10" />);

const CheckCircle = createIcon(() => (
  <Fragment>
    <path d="M22 11.08V12a10 10 0 1 1-5.93-9.14" />
    <polyline points="22 4 12 14.01 9 11.01" />
  </Fragment>
));

const XCircle = createIcon(() => (
  <Fragment>
    <circle cx="12" cy="12" r="10" />
    <line x1="15" y1="9" x2="9" y2="15" />
    <line x1="9" y1="9" x2="15" y2="15" />
  </Fragment>
));

const FilledCircle = createIcon(() => (
  <circle fill="currentColor" cx="12" cy="12" r="10" />
));

const ContextMenuIcon = createIcon(() => (
  <Fragment>
    <circle fill="currentColor" cx="12" cy="12" r="1" />
    <circle fill="currentColor" cx="19" cy="12" r="1" />
    <circle fill="currentColor" cx="5" cy="12" r="1" />
  </Fragment>
));

const Zap = createIcon(() => (
  <polygon points="13 2 3 14 12 14 11 22 21 10 12 10 13 2" />
));

export {
  Icon,
  Home,
  Folder,
  File,
  IdFile,
  Circle,
  CheckCircle,
  XCircle,
  IconWrapper,
  IconSvg,
  FilledCircle,
  ContextMenuIcon,
  Zap,
};
