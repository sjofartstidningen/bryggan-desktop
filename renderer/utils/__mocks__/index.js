const createSimpleCache = () => {
  const cache = new Map();

  return {
    set: jest.fn((key, value) => cache.set(key, value)),
    get: jest.fn(key => cache.get(key)),
    has: jest.fn(key => cache.has(key)),
    delete: jest.fn(key => cache.delete(key)),
    clear: jest.fn(() => cache.clear()),
  };
};

module.exports = { createSimpleCache };
