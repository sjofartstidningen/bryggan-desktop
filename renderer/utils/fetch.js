const fetch = window && window.fetch ? window.fetch : require('node-fetch');

const AbortController =
  window && window.AbortController
    ? window.AbortController
    : require('abort-controller');

export { fetch, AbortController };
