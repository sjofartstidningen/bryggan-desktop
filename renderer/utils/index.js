import { extname } from 'path';

const sortByType = (a, b) => {
  if (a.type !== b.type) return a.type > b.type ? -1 : 1;

  const aExt = extname(a.name);
  const bExt = extname(b.name);
  if (aExt !== bExt) return aExt > bExt ? 1 : -1;
  if (a.name > b.name) return 1;
  if (a.name < b.name) return -1;
  return 0;
};

export { sortByType };
