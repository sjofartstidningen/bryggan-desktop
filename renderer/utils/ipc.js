import log from 'electron-log';

const callMain = async (...args) => {
  const ipc = require('electron-better-ipc');
  log.verbose(`Call main on channel: ${args[0]}`);
  return ipc.callMain(...args);
};

const answerMain = (...args) => {
  const ipc = require('electron-better-ipc');
  log.verbose(`Setup answer to main on channel: ${args[0]}`);
  return ipc.answerMain(...args);
};

export { callMain, answerMain };
