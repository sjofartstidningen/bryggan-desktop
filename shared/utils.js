import path from 'path';

const hasIdlkFile = (file, items) => {
  /**
   * .idlk files are special in that the are somewhat similar to the original
   * filename â€“ but not exactly
   * The original filename is capped at 18 chars and transformed to lower case
   * then wrapped in tildes: "~[original_filename]~[random_chars].idlk"
   */
  const basename = path
    .basename(file.name, '.indd')
    .slice(0, 18)
    .toLowerCase();

  for (const item of items) {
    if (path.extname(item.name) === '.idlk') {
      const [, strippedIdlkName] = item.name.match(/~(.+)~/);
      if (strippedIdlkName === basename) return true;
    }
  }

  return false;
};

export { hasIdlkFile };
