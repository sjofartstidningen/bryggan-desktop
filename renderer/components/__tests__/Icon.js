import React from 'react';
import { renderAndMatch } from '../../../tests/utils';
import * as Icon from '../Icon';

Object.entries(Icon).forEach(([name, Comp]) => {
  describe(`Component: <Icon.${name} />`, () => {
    if (name === 'Icon') {
      renderAndMatch(
        <Comp>
          <circle cx="12" cy="12" r="10" />
        </Comp>,
      );
    } else {
      renderAndMatch(<Comp />);
    }
  });
});
