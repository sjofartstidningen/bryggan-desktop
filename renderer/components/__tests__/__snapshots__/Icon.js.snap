// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <Icon.File /> should render without issues 1`] = `
.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  width: 1em;
  height: 1em;
}

.c1 {
  width: 1em;
  height: 1em;
  fill: none;
  stroke: currentColor;
  stroke-width: 2;
  stroke-linecap: round;
  stroke-linejoin: round;
  position: absolute;
  bottom: -0.125em;
}

<span
  class="c0"
>
  <svg
    class="c1"
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M13 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V9z"
    />
    <polyline
      points="13 2 13 9 20 9"
    />
  </svg>
</span>
`;

exports[`Component: <Icon.Folder /> should render without issues 1`] = `
.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  width: 1em;
  height: 1em;
}

.c1 {
  width: 1em;
  height: 1em;
  fill: none;
  stroke: currentColor;
  stroke-width: 2;
  stroke-linecap: round;
  stroke-linejoin: round;
  position: absolute;
  bottom: -0.125em;
}

<span
  class="c0"
>
  <svg
    class="c1"
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M22 19a2 2 0 0 1-2 2H4a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h5l2 3h9a2 2 0 0 1 2 2z"
    />
  </svg>
</span>
`;

exports[`Component: <Icon.Home /> should render without issues 1`] = `
.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  width: 1em;
  height: 1em;
}

.c1 {
  width: 1em;
  height: 1em;
  fill: none;
  stroke: currentColor;
  stroke-width: 2;
  stroke-linecap: round;
  stroke-linejoin: round;
  position: absolute;
  bottom: -0.125em;
}

<span
  class="c0"
>
  <svg
    class="c1"
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M3 9l9-7 9 7v11a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2z"
    />
    <polyline
      points="9 22 9 12 15 12 15 22"
    />
  </svg>
</span>
`;

exports[`Component: <Icon.Icon /> should render without issues 1`] = `
.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  width: 1em;
  height: 1em;
}

.c1 {
  width: 1em;
  height: 1em;
  fill: none;
  stroke: currentColor;
  stroke-width: 2;
  stroke-linecap: round;
  stroke-linejoin: round;
  position: absolute;
  bottom: -0.125em;
}

<span
  class="c0"
>
  <svg
    class="c1"
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      cx="12"
      cy="12"
      r="10"
    />
  </svg>
</span>
`;

exports[`Component: <Icon.IconSvg /> should render without issues 1`] = `
.c0 {
  width: 1em;
  height: 1em;
  fill: none;
  stroke: currentColor;
  stroke-width: 2;
  stroke-linecap: round;
  stroke-linejoin: round;
}

<svg
  class="c0"
/>
`;

exports[`Component: <Icon.IconWrapper /> should render without issues 1`] = `
.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  width: 1em;
  height: 1em;
}

<span
  class="c0"
/>
`;

exports[`Component: <Icon.IdFile /> should render without issues 1`] = `
.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  width: 1em;
  height: 1em;
}

.c1 {
  width: 1em;
  height: 1em;
  fill: none;
  stroke: currentColor;
  stroke-width: 2;
  stroke-linecap: round;
  stroke-linejoin: round;
  position: absolute;
  bottom: -0.125em;
}

<span
  class="c0"
>
  <svg
    class="c1"
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M13 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V9z"
    />
    <path
      d="M9.57,16c0,.08,0,.1-.1.1H8.32c-.08,0-.1,0-.1-.1V7.91c0-.07,0-.1.1-.1H9.48a.08.08,0,0,1,.09.09Z"
      fill="#000"
      stroke-width="0"
    />
    <path
      d="M15.75,14.9v-7s0-.06-.08-.06H14.45a.07.07,0,0,0-.07.08v2H14a3,3,0,0,0-3.08,3.2h0c0,2.09,1.19,3.12,2.85,3.12a4.59,4.59,0,0,0,2-.45s.06,0,.06-.08A7.15,7.15,0,0,1,15.75,14.9Zm-1.37-3.8v3.8a1.53,1.53,0,0,1-.54.09c-.86,0-1.55-.57-1.55-2,0-1.24.67-2,1.64-2a1.13,1.13,0,0,1,.45.08Z"
      fill="#000"
      stroke-width="0"
    />
  </svg>
</span>
`;

exports[`Component: <Icon.XCircle /> should render without issues 1`] = `
.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  width: 1em;
  height: 1em;
}

.c1 {
  width: 1em;
  height: 1em;
  fill: none;
  stroke: currentColor;
  stroke-width: 2;
  stroke-linecap: round;
  stroke-linejoin: round;
  position: absolute;
  bottom: -0.125em;
}

<span
  class="c0"
>
  <svg
    class="c1"
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      cx="12"
      cy="12"
      r="10"
    />
    <line
      x1="15"
      x2="9"
      y1="9"
      y2="15"
    />
    <line
      x1="9"
      x2="15"
      y1="9"
      y2="15"
    />
  </svg>
</span>
`;
